packages/core/src/external/opensearch/file-search-connector-factory.ts

```diff
@@ -1,16 +1,18 @@
+import { getSecret } from "@aws-lambda-powertools/parameters/secrets";
 import { OpenSearchFileIngestor } from "./file-ingestor";
 import { OpenSearchFileIngestorDirect } from "./file-ingestor-direct";
 import { OpenSearchFileIngestorSQS } from "./file-ingestor-sqs";
 import { OpenSearchFileSearcher } from "./file-searcher";
 import { OpenSearchFileSearcherDirect } from "./file-searcher-direct";
 import { Config } from "../../util/config";
 
-export function makeSearchServiceIngest(): OpenSearchFileIngestor {
+export async function makeSearchServiceIngest(): Promise<OpenSearchFileIngestor> {
```

Comment:
had to make these async because getSecret is async: https://aws.amazon.com/blogs/compute/retrieving-parameters-and-secrets-with-powertools-for-aws-lambda-typescript/